Move namespace of adapter to active model serializers

domitian opened the pull request:
#1446 will be fixed. Adapter moved from serializers to active_model_serializers namespace.
@bf4 need your help rebasing commits and coderefactoring.

domitian commented:
Should I make code changes to make it pass for rubocop as well?

bf4 commented:
Yup :)

Thanks!

B mobile phone

> On Feb 6, 2016, at 10:39 AM, domitian <notifications@github.com> wrote:
> 
> Should I make code changes to make it pass for rubocop as well?
> 
> —
> Reply to this email directly or view it on GitHub.
>

domitian commented:
@bf4 all ci checks are fine.
But one thing I noticed is why didn't rubocop throw these offences when adapter is under ActiveModel::Serializer.
Then I noticed that the files which I made changes for are ignored in .rubocop_todo.yml here https://github.com/rails-api/active_model_serializers/blob/master/.rubocop_todo.yml#L16#L24
Since I made code changes to pass the CI instead of changed rubocop config, I don't know if it's the right thing to do. Please help on this.

bf4 commented:
@domitian Thanks for this.  I'm going to look it over a bit later.  Also, heads up, I'd like to merge https://github.com/rails-api/active_model_serializers/pull/1471 first and keep in mind the effect of this on other active adapter-specific prs...

bf4 commented on lib/active_model/serializable_resource.rb:
```diff
@@ -3,6 +3,7 @@ module ActiveModel
   class SerializableResource
     ADAPTER_OPTION_KEYS = Set.new([:include, :fields, :adapter, :meta, :meta_key, :links])
     include ActiveModelSerializers::Logging
+    require 'active_model_serializers/adapter'
```
any require you put the require here and not at the top?

bf4 commented:
Looks good. Your thoughts re: https://github.com/rails-api/active_model_serializers/pull/1499#issuecomment-181102979

bf4 commented on lib/active_model/serializer.rb:
```diff
@@ -21,7 +21,7 @@ class Serializer
     include Links
     include Type
-    require 'active_model/serializer/adapter'
+    require 'active_model_serializers/adapter'
```
I think this should be removed now, along with the `adapter` method (which I guess we can keep around with a deprecation warnings)

domitian replied:
I removed the adapter dependency and added adapter method with deprecation warning. To replace `adapter` method, I created a similar one in Adapter module called `lookup_from_config`. Can you look at the changes I made  in the separate branch I made https://github.com/domitian/active_model_serializers/compare/move-namespace-of-adapter-to-active-model-serializers...domitian:removing_adapter_from_serializer#diff-5849afc4f57a4ff11910f672112e293aR16

bf4 replied:
I think it's good. It has some extra spacing in one of the methods.

Maybe call it `configured_adapter`?

domitian replied:
that seems to be a good name.

domitian commented:
@bf4 
Made the namespace changes in docs as well, and added deprecation warning/message to requiring adapter inside serializer, renamed and added the `adapter` method in serializer to Adapter::configured_adapter.

I believe I made all the changes that are necessary for this PR, do you think are there any changes that needs to be done before I merge latest master into this branch(Seems to be lot of changes yesterday)?

remear commented on docs/general/adapters.md:
```diff
@@ -117,46 +117,46 @@ The default adapter can be configured, as above, to use any class given to it.
 
-1) The simplest, is to subclass `ActiveModel::Serializer::Adapter::Base`, e.g. the below will
+1) The simplest, is to subclass `ActiveModelSerializers::Adapter::Base`, e.g. the below will
 register the `Example::UsefulAdapter` as `"example/useful_adapter"`.
```
Did you want example/useful_adapter quoted in the inline code block, or is just the code block sufficient in this context?  `"example/useful_adapter"` vs `example/useful_adapter`

bf4 replied:
It's  `"example/useful_adapter"`, though it can be looked up as `:"example/useful_adapter" if you want

```ruby
# code that registers the adapter `name` to the `klass`
 name = name.to_s.gsub(/\AActiveModel::Serializer::Adapter::/, ''.freeze)
 adapter_map.update(name.underscore => klass)
```

remear commented on lib/active_model/serializable_resource.rb:
```diff
@@ -1,4 +1,5 @@
 require 'set'
+require 'active_model_serializers/adapter'
```
Line return after this?

remear commented on test/adapter/json_api/parse_test.rb:
```diff
@@ -1,136 +1,134 @@
 require 'test_helper'
```
Line return here?

remear commented:
Added a few comments. This looks good to me. :+1:
